### ISSUE TITLE
Touching WORKSPACE and building //... fails

### ISSUE THREAD

**Issue by kchodorow** [bazelbuild/bazel#421]
@pgr0ss noticed that if someone creates a new workspace, touches the WORKSPACE file, and runs //..., it will fail because the base_workspace tools now have a bunch of external dependencies from skylark rules (https://github.com/bazelbuild/bazel/blob/master/tools/BUILD#L28-L30) that aren't declared in the user's WORKSPACE file. Maybe all BUILD rules with external dependencies should be declared in the appengine.BUILD (or rust.BUILD) files so they don't get dragged into the user's base workspace.

**Comment by damienmg** [comment]
We have a long term plan (ETA ~2month) for that (ship the tools and skylark rules through remote repositories). Until then I don't think it worth doing anything.

**Comment by pgr0ss** [comment]
@damienmg If I understand correctly, that means that we can't use the packaged release candidate and `bazel build //...` or `bazel test //...` to build and test our app for the next two months? I think this means we'll have to stick with using bazel from a `git clone`.

**Comment by kchodorow** [comment]
It's actually not limited to the binary release, the git clone will have
the same issue.

**Comment by pgr0ss** [comment]
We currently use the git clone and don't have any issues. We only had a problem when we tried to switch from the clone to the packaged released candidate.

**Comment by dionescu** [comment]
I can confirm that bazel build //... works with a fresh version of bazel obtained via git clone. It does not work with the release binary.

If someone from Kythe sees this, maybe they can update their build from source instructions.

**Comment by damienmg** [comment]
//... doesn't recurse into symlinks. As a consequence, the way we install when using compile.sh in which the `tools` folder is a symlink prevent things under tools from being executed while the installer does copy so no symlinks.

**Comment by damienmg** [comment]
Reassigning to Lukacs who started working on shiping the tools directory.

**Comment by pspeter3** [comment]
Using `bazel build //...` with a fresh `git clone` no longer works nor does it work in the packaged release

**Comment by lberki** [comment]
See this thread:
https://groups.google.com/d/msg/bazel-discuss/GuS_QfVh4l0/42zB0wpVAgAJ

TL;DR: Sorry, will be fixed real soon now. For now, the workaround is
"bazel build -- //... -//tools/..."

On Tue, Oct 27, 2015 at 3:42 AM, Phips Peter <notifications@github.com>
wrote:

> Using bazel build //... with a fresh git clone no longer works nor does
> it work in the packaged release
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/bazelbuild/bazel/issues/421#issuecomment-151352180>.
>



-- 
Lukács T. Berki | Software Engineer | lberki@google.com |

Google Germany GmbH | Maximillianstr. 11-15 | 80539 München | Germany
| Geschäftsführer: Matthew
Scott Sucherman, Paul Terence Manicle | Registergericht und -nummer:
Hamburg, HRB 86891

**Comment by pspeter3** [comment]
Is there not a safe version of the base_workspace that only has the dependencies in the build encyclopedia?

**Comment by lberki** [comment]
Even if there was, there would still be a problem that iOS tools cannot be
built on Linux, so it wouldn't help you much. You can try deleting
tools/build_defs and tools/build_rules and I think that will make things a
bit more palatable.

On Wed, Oct 28, 2015 at 1:10 AM, Phips Peter <notifications@github.com>
wrote:

> Is there not a safe version of the base_workspace that only has the
> dependencies in the build encyclopedia?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/bazelbuild/bazel/issues/421#issuecomment-151680693>.
>



-- 
Lukács T. Berki | Software Engineer | lberki@google.com |

Google Germany GmbH | Maximillianstr. 11-15 | 80539 München | Germany
| Geschäftsführer: Matthew
Scott Sucherman, Paul Terence Manicle | Registergericht und -nummer:
Hamburg, HRB 86891

**Comment by kchodorow** [comment]
This should now work (fixed by e87849b8d391af8f5d98e3a91e680e88a1264b64).
